### PR TITLE
set permissions and group on directories to sync

### PR DESCRIPTION
**Pull request by b97pla**
- set "ngi" as group on `ngi_containers` and `wildest`
- set full permissions for group and remove permissions for others
- do not try to set owner after sync

### CODE REVIEW

File: group_vars/all/main.yml [hunk 1]
```diff
@@ -8,5 +8,6 @@ deployment_static_resources: "{{ deployment_remote_path }}/deploy/files"
 
 static_proj_root: /proj
 
-ngi_group: ngi-sw
+ngi_sw_group: ngi-sw
```
Comment (mariya) [inline group_vars/all/main.yml:11]: Is there a reason to have these as variables if the variable names are identical to their values?

File: install.yml [hunk 2]
```diff
@@ -65,18 +65,26 @@
     - name: set correct file permission for everything in the deployment
       file:
         path: "{{ root_path }}"
+        group: "{{ ngi_sw_group }}"
+        state: directory
+        recurse: yes
+        mode: g+rwXs,o=rX
```
Comment (mariya) [inline install.yml:71]: Don't we want to remove all permissions for others? (o-rwx)

File: sync.yml [hunk 3]
```diff
@@ -23,7 +23,7 @@
   tags: permissions
 
   vars:
-    skip_permission_check: false
+    skip_permission_check: true
```
Comment (mariya) [inline sync.yml:26]: What does this change and what are the implications of that?

**Review by mariya — CHANGES_REQUESTED** [review]
A few comments/questions... I also think this should be tested before merging.

File: group_vars/all/main.yml (hunk 1 above)
Comment (b97pla) [reply group_vars/all/main.yml:11]: maybe not I guess, although I'm in favor of using variables out of principle..

File: install.yml (hunk 2 above)
Comment (b97pla) [reply install.yml:71]: no, others need to be able to access stuff under this hierarchy in order to use the deployment

File: sync.yml (hunk 3 above)
Comment (b97pla) [reply sync.yml:26]: it just skip the checking of permissions below by default. The permissions are set explicitly at the end of the `install.yml` play, so I don't think this step is necessary.

if permissions turn out to be a problem in practice down the line, we can consider doing this differently

**Comment by b97pla** [comment]
Right, thanks. I'll try to test this properly.

File: sync.yml (hunk 3 above)
Comment (b97pla) [reply sync.yml]: I have now changed this behavior so that the permission check is tagged and can be skipped with `--skip-tags` if needed

**Comment by b97pla** [comment]
Ok, I've made some further changes and tested this as far as possible without actually syncing the data to the cluster. That is, after deployment in development environment, the sync playbook has been run with `--dry-run` for rsync.

File: install.yml [hunk 4]
```diff
@@ -65,19 +94,18 @@
...
       file:
-        path: "{{ proj_root }}"
-        group: "{{ ngi_group }}"
+        path: "{{ ngi_containers }}"
+        group: "{{ ngi_sw_group }}"
         state: directory
         recurse: yes
-        mode: g+rwXs,o=rwX
-      when: deployment_environment == "staging"
+        mode: g+rwX,o=rX
```
Comment (GitEdvard) [inline install.yml:108]: I see that the 's' flag is removed from the recursive mode assignment, and this 's' flag seems to be applied only on the directory root folder. Is there perhaps any chance that there could be an already existing sub-folder that doesn't have this setting?

File: roles/func_accounts/meta/main.yml [hunk 5]
```diff
@@ -1,2 +1,2 @@
 dependencies:
-  - { role: anaconda, tags: anaconda }
+  - { role: ngi_pipeline, tags: ngi_pipeline }
```
Comment (GitEdvard) [inline roles/func_accounts/meta/main.yml:2]: I wonder, does it ever happen that one runs a specific role from miarka-provision? It seems to me that the entire install.yml playbook is run as a whole each time.

**Review by GitEdvard — APPROVED** [review]
Great work!

File: install.yml (hunk 4 above)
Comment (b97pla) [reply install.yml:108]: this is a great point, right now, there is indeed some existing sub-folders and I intend to deal with them when this is to be deployed and synced for the first time, either by (re-)moving offending folders or asking Uppmax to update ownership/permissions. thereafter, it is expected that everything is either only added through this playbook + sync or will inherit the root folder settings + effective umask of user (the plan is that Uppmax will set a global umask). We were also discussing options for ensuring the consistency of permissions and ownership in the projects jus this morning..

**Pull request merged**